plugins/imfile/imfile.c

```diff
@@ -1374,8 +1374,9 @@ enqLine(act_obj_t *const act,
 		size_t ceeMsgSize = msgLen + CONST_LEN_CEE_COOKIE + 1;
 		char *ceeMsg;
 		CHKmalloc(ceeMsg = malloc(ceeMsgSize));
-		strcpy(ceeMsg, CONST_CEE_COOKIE);
-		strcat(ceeMsg, (char*)rsCStrGetSzStrNoNULL(cstrLine));
+		snprintf(ceeMsg, ceeMsgSize, "%s%s", CONST_CEE_COOKIE,
```

Comment:
I don't like snprintf here - it is so much slower. What is the issue with strcpy at this point inside the code? I guess it is more the bounding of the strcat(), right? If so, it probably makes sense to query cstrLine for its size - but this is done in line 1362 already, so we know the buffer is sufficiently large.  We could probably replace the strcpy() by a memcpy() if we fear that an attacker might change the buffer between the point where the size is taken and then used. What do you think?